plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/HeteroDescribableConfigurator.java

```diff
@@ -133,8 +136,47 @@ public CNode describe(T instance, ConfigurationContext context) {
                 .getOrElseThrow(() -> new IllegalStateException("No configurator implementation to manage " + klazz));
     }
 
+    /**
+     * Matches suitable descriptors for the target.
+     * The fetch is trivial when the target implements a root {@link Describable} object.
+     * If not, we iterate to parent classes until we find a class which can provide the descriptor list in {@link Jenkins#getDescriptorList(Class)}.
+     * Then we go through all the descriptors and find ones compliant with the target.
+     * @return Stream of descriptors which match the target
+     */
     private Stream<Descriptor<T>> getDescriptors() {
-        return Stream.ofAll(Jenkins.getInstance().getDescriptorList(target));
+        DescriptorExtensionList<T, Descriptor<T>> descriptorList = Jenkins.getInstance().getDescriptorList(target);
+        if (!descriptorList.isEmpty()) { // fast fetch for root objects
+            return Stream.ofAll(descriptorList);
+        }
+
+        DescriptorExtensionList parentDescriptorClassList = descriptorList;
+        Class<?> effectiveTarget = target;
```

Comment:
```suggestion
        Class<?> effectiveTarget = target.getSuperclass();
```

Follow-up to https://github.com/jenkinsci/configuration-as-code-plugin/pull/932/files#r297513382 from @fcojfernandez